### ISSUE TITLE
Cannot start HDT after installing Win 64 b2004

### ISSUE THREAD

**Issue by aethelstan2** [HearthSim/Hearthstone-Deck-Tracker#4120]
### Expected Behavior
HDT to start up

### Actual Behavior
no startup. no error message from the app. I have to search windows events logs

### Steps to reproduce behavior
Simply start the app

### Log/Screenshots

```
- <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
- <System>
  <Provider Name=".NET Runtime" /> 
  <EventID Qualifiers="0">1026</EventID> 
  <Version>0</Version> 
  <Level>2</Level> 
  <Task>0</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2020-07-16T16:14:36.2766991Z" /> 
  <EventRecordID>453</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="0" ThreadID="0" /> 
  <Channel>Application</Channel> 
  <Computer>PC-LOLO</Computer> 
  <Security /> 
  </System>
- <EventData>
  <Data>Application : HearthstoneDeckTracker.exe Version du Framework : v4.0.30319 Description : le processus a été arrêté en raison d'une exception non gérée. Informations sur l'exception : System.ArithmeticException Informations sur l'exception : System.OverflowException à System.Windows.Media.Imaging.ColorConvertedBitmap.FinalizeCreation() à System.Windows.Media.Imaging.ColorConvertedBitmap..ctor(System.Windows.Media.Imaging.BitmapSource, System.Windows.Media.ColorContext, System.Windows.Media.ColorContext, System.Windows.Media.PixelFormat) à System.Windows.Media.Imaging.BitmapSource.CreateCachedBitmap(System.Windows.Media.Imaging.BitmapFrame, System.Windows.Media.Imaging.BitmapSourceSafeMILHandle, System.Windows.Media.Imaging.BitmapCreateOptions, System.Windows.Media.Imaging.BitmapCacheOption, System.Windows.Media.Imaging.BitmapPalette) à System.Windows.Media.Imaging.BitmapFrameDecode.FinalizeCreation() à System.Windows.Media.Imaging.BitmapFrameDecode..ctor(Int32, System.Windows.Media.Imaging.BitmapCreateOptions, System.Windows.Media.Imaging.BitmapCacheOption, System.Windows.Media.Imaging.BitmapDecoder) à System.Windows.Media.Imaging.BitmapDecoder.SetupFrames(System.Windows.Media.Imaging.BitmapDecoder, System.Collections.ObjectModel.ReadOnlyCollection`1<System.Windows.Media.Imaging.BitmapFrame>) à System.Windows.Media.Imaging.BitmapDecoder.get_Frames() à System.Windows.Media.Imaging.BitmapFrame.CreateFromUriOrStream(System.Uri, System.Uri, System.IO.Stream, System.Windows.Media.Imaging.BitmapCreateOptions, System.Windows.Media.Imaging.BitmapCacheOption, System.Net.Cache.RequestCachePolicy) à System.Windows.Media.ImageSourceConverter.ConvertFrom(System.ComponentModel.ITypeDescriptorContext, System.Globalization.CultureInfo, System.Object) à MS.Internal.Xaml.Runtime.ClrObjectRuntime.CreateObjectWithTypeConverter(MS.Internal.Xaml.ServiceProviderContext, System.Xaml.Schema.XamlValueConverter`1<System.ComponentModel.TypeConverter>, System.Object) à MS.Internal.Xaml.Runtime.ClrObjectRuntime.CreateFromValue(MS.Internal.Xaml.ServiceProviderContext, System.Xaml.Schema.XamlValueConverter`1<System.ComponentModel.TypeConverter>, System.Object, System.Xaml.XamlMember) à MS.Internal.Xaml.Runtime.PartialTrustTolerantRuntime.CreateFromValue(MS.Internal.Xaml.ServiceProviderContext, System.Xaml.Schema.XamlValueConverter`1<System.ComponentModel.TypeConverter>, System.Object, System.Xaml.XamlMember) à System.Xaml.XamlObjectWriter.Logic_CreateFromValue(MS.Internal.Xaml.Context.ObjectWriterContext, System.Xaml.Schema.XamlValueConverter`1<System.ComponentModel.TypeConverter>, System.Object, System.Xaml.XamlMember, System.String, MS.Internal.Xaml.Runtime.IAddLineInfo) Informations sur l'exception : System.Windows.Markup.XamlParseException à System.Windows.Markup.WpfXamlLoader.Load(System.Xaml.XamlReader, System.Xaml.IXamlObjectWriterFactory, Boolean, System.Object, System.Xaml.XamlObjectWriterSettings, System.Uri) à System.Windows.Markup.WpfXamlLoader.LoadBaml(System.Xaml.XamlReader, Boolean, System.Object, System.Xaml.Permissions.XamlAccessLevel, System.Uri) à System.Windows.Markup.XamlReader.LoadBaml(System.IO.Stream, System.Windows.Markup.ParserContext, System.Object, Boolean) à System.Windows.Application.LoadComponent(System.Object, System.Uri) à Hearthstone_Deck_Tracker.App.InitializeComponent() à Hearthstone_Deck_Tracker.App.Main()</Data> 
  </EventData>
  </Event>

```

```
Application : HearthstoneDeckTracker.exe 
Version du Framework : v4.0.30319 
Description : le processus a été arrêté en raison d'une exception non gérée. 
Informations sur l'exception : System.ArithmeticException 
Informations sur l'exception : System.OverflowException 
à System.Windows.Media.Imaging.ColorConvertedBitmap.FinalizeCreation() 
à System.Windows.Media.Imaging.ColorConvertedBitmap..ctor(System.Windows.Media.Imaging.BitmapSource, System.Windows.Media.ColorContext, System.Windows.Media.ColorContext, System.Windows.Media.PixelFormat) 
à System.Windows.Media.Imaging.BitmapSource.CreateCachedBitmap(System.Windows.Media.Imaging.BitmapFrame, System.Windows.Media.Imaging.BitmapSourceSafeMILHandle, System.Windows.Media.Imaging.BitmapCreateOptions, System.Windows.Media.Imaging.BitmapCacheOption, System.Windows.Media.Imaging.BitmapPalette) 
à System.Windows.Media.Imaging.BitmapFrameDecode.FinalizeCreation() 
à System.Windows.Media.Imaging.BitmapFrameDecode..ctor(Int32, System.Windows.Media.Imaging.BitmapCreateOptions, System.Windows.Media.Imaging.BitmapCacheOption, System.Windows.Media.Imaging.BitmapDecoder) 
à System.Windows.Media.Imaging.BitmapDecoder.SetupFrames(System.Windows.Media.Imaging.BitmapDecoder, System.Collections.ObjectModel.ReadOnlyCollection`1<System.Windows.Media.Imaging.BitmapFrame>) 
à System.Windows.Media.Imaging.BitmapDecoder.get_Frames() 
à System.Windows.Media.Imaging.BitmapFrame.CreateFromUriOrStream(System.Uri, System.Uri, System.IO.Stream, System.Windows.Media.Imaging.BitmapCreateOptions, System.Windows.Media.Imaging.BitmapCacheOption, System.Net.Cache.RequestCachePolicy) 
à System.Windows.Media.ImageSourceConverter.ConvertFrom(System.ComponentModel.ITypeDescriptorContext, System.Globalization.CultureInfo, System.Object) y
à MS.Internal.Xaml.Runtime.ClrObjectRuntime.CreateObjectWithTypeConverter(MS.Internal.Xaml.ServiceProviderContext, System.Xaml.Schema.XamlValueConverter`1<System.ComponentModel.TypeConverter>, System.Object) 
à MS.Internal.Xaml.Runtime.ClrObjectRuntime.CreateFromValue(MS.Internal.Xaml.ServiceProviderContext, System.Xaml.Schema.XamlValueConverter`1<System.ComponentModel.TypeConverter>, System.Object, System.Xaml.XamlMember) 
à MS.Internal.Xaml.Runtime.PartialTrustTolerantRuntime.CreateFromValue(MS.Internal.Xaml.ServiceProviderContext, System.Xaml.Schema.XamlValueConverter`1<System.ComponentModel.TypeConverter>, System.Object, System.Xaml.XamlMember) 
à System.Xaml.XamlObjectWriter.Logic_CreateFromValue(MS.Internal.Xaml.Context.ObjectWriterContext, System.Xaml.Schema.XamlValueConverter`1<System.ComponentModel.TypeConverter>, System.Object, System.Xaml.XamlMember, System.String, MS.Internal.Xaml.Runtime.IAddLineInfo)
Informations sur l'exception : System.Windows.Markup.XamlParseException 
à System.Windows.Markup.WpfXamlLoader.Load(System.Xaml.XamlReader, System.Xaml.IXamlObjectWriterFactory, Boolean, System.Object, System.Xaml.XamlObjectWriterSettings, System.Uri) 
à System.Windows.Markup.WpfXamlLoader.LoadBaml(System.Xaml.XamlReader, Boolean, System.Object, System.Xaml.Permissions.XamlAccessLevel, System.Uri) 
à System.Windows.Markup.XamlReader.LoadBaml(System.IO.Stream, System.Windows.Markup.ParserContext, System.Object, Boolean) 
à System.Windows.Application.LoadComponent(System.Object, System.Uri) 
à Hearthstone_Deck_Tracker.App.InitializeComponent() 
à Hearthstone_Deck_Tracker.App.Main()
```

AND

```

- <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
- <System>
  <Provider Name="Application Error" /> 
  <EventID Qualifiers="0">1000</EventID> 
  <Version>0</Version> 
  <Level>2</Level> 
  <Task>100</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2020-07-16T16:14:36.3391842Z" /> 
  <EventRecordID>454</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="0" ThreadID="0" /> 
  <Channel>Application</Channel> 
  <Computer>PC-LOLO</Computer> 
  <Security /> 
  </System>
- <EventData>
  <Data>HearthstoneDeckTracker.exe</Data> 
  <Data>1.12.16.4032</Data> 
  <Data>5f0e21ca</Data> 
  <Data>KERNELBASE.dll</Data> 
  <Data>10.0.19041.388</Data> 
  <Data>a2224bb6</Data> 
  <Data>e0434352</Data> 
  <Data>00129862</Data> 
  <Data>9f0</Data> 
  <Data>01d65b8c32efaf03</Data> 
  <Data>C:\Users\Laurent\AppData\Local\HearthstoneDeckTracker\app-1.12.16\HearthstoneDeckTracker.exe</Data> 
  <Data>C:\WINDOWS\System32\KERNELBASE.dll</Data> 
  <Data>02ef3702-2d26-4bed-b7d5-db93dbce892d</Data> 
  <Data /> 
  <Data /> 
  </EventData>
  </Event>
```

**Comment by rafiqm** [comment]
Experiencing same issue, and I've found a workaround in the source.

- Occurs on current version **1.12.16** (via installer)
- Occurs on **1.12.13** (via github download release)
- Started occuring directly after updating to latest version of Windows 10 (v2004, build 19041.264) - used it a few mins before update and there was no issue.
- After building source successfully and running, I receive the following error:
`System.Windows.Markup.XamlParseException: ''Failed to create a 'ImageSource' from the text 'Resources/dust.png'.' Line number '39' and line position '22'.'
`
- I'm able to open Resources/dust.png in Windows image viewer and it looks fine
- Running pngcheck on dust.png does not show any problems
- **Workaround:** If I remove Resources/dust.png and copy another image in it's place (e.g. damage.png -> dust.png) then program starts fine

That's the limits of my debugging I'm afraid, I'm not a .NET developer, so I can't investigate further or submit a PR. That said, I don't know whether it's a code problem anyway, it looks just like it's something wrong with that specific PNG.

[HDT-Exception-VisualStudio-20200720.txt](https://github.com/HearthSim/Hearthstone-Deck-Tracker/files/4945772/HDT-Exception-VisualStudio-20200720.txt)

[HDTErrors-20200720.txt](https://github.com/HearthSim/Hearthstone-Deck-Tracker/files/4945356/HDTErrors-20200720.txt)

**Comment by aethelstan2** [comment]
@rafiqm ty so much for you help.
@riQQ could you please build-up a new installer with a new picture of dust or tell me how to compile the sources right there ?

**Comment by azeier** [comment]
Well this is a weird bug!

I tried running optipng over the dust image as a start. Can you please try running this build: https://media.hearthsim.net/uploads/HDT-Test-v1.12.16-1.zip

I'm currently sadly unable to install Win10 2004, so this will need to a bit of trial and error.

**Comment by aethelstan2** [comment]
@azeier Hi !
It works :) 
Could you please compile a new installer ? :P 

ty,

**Comment by azeier** [comment]
That's great news!

Here's an installer dev build to make sure this all works there too: https://github.com/HearthSim/HDT-dev-builds/releases/download/v1.12.16.4033/HDT-Installer.exe
This will overwrite your default installation and automatically update to the normal release once it's available

**Comment by rafiqm** [comment]
> Here's an installer dev build to make sure this all works there too: https://github.com/HearthSim/HDT-dev-builds/releases/download/v1.12.16.4033/HDT-Installer.exe

Confirmed that this fixes the issue for me too, thanks for the quick action.

**Comment by aethelstan2** [comment]
All good. Thank a lot :)

**Comment by azeier** [comment]
Thanks for the help and testing it guys!

This should now be fixed with v1.12.17.
